netbox/netbox/jobs.py

```diff
@@ -2,19 +2,38 @@
 from abc import ABC, abstractmethod
 from datetime import timedelta
 
+from django.core.exceptions import ImproperlyConfigured
 from django.utils.functional import classproperty
 from django_pglocks import advisory_lock
 from rq.timeouts import JobTimeoutException
 
 from core.choices import JobStatusChoices
 from core.models import Job, ObjectType
 from netbox.constants import ADVISORY_LOCK_KEYS
+from netbox.registry import registry
 
 __all__ = (
     'JobRunner',
+    'system_job',
 )
 
 
+def system_job(interval):
```

Comment:
I suggest adding `*args, **kwargs` as arguments, storing them in the registry and passing them to `enqueue_once()`. This would allow plugins to reuse jobs for different actions. Practical example: Changelog and journal cleanup could use the same job and pass model/configuration parameter names as arguments.

Reply:
We want to avoid caching arbitrary data in the registry, as it needs to be constructed (consuming memory) for each running process. Creating multiple subclasses of a base Job class is probably a more scalable approach.